### PR TITLE
Include what you use + more SPDX identifiers + few other cleanups

### PR DESCRIPTION
**Pull request by piponazo**
The main objective of this PR was to cleanup the list of header inclusions in our source files thanks to the usage of the tool "Include what you use":
https://github.com/include-what-you-use/include-what-you-use

While doing these changes I noticed that I forgot to use the SPDX identifiers in some source files (see #2122). 

Apart from that there are few other small changes.

### CODE REVIEW

File: include/exiv2/error.hpp [hunk 1]
```diff
@@ -14,8 +14,9 @@
 // *****************************************************************************
 #include "exiv2lib_export.h"
 
-// included header files
-#include "types.hpp"
+#include <bits/exception.h>   // for exception
```
Comment (neheb) [inline include/exiv2/error.hpp]: This is an internal libc++ header. Use <exception>

File: src/image_int.hpp [hunk 2]
```diff
@@ -5,9 +5,11 @@
 
 // *****************************************************************************
 // included header files
-#include "types.hpp"
+#include "slice.hpp"  // for Slice
 
-// + standard includes
+#include <stddef.h>   // for size_t
```
Comment (neheb) [inline src/image_int.hpp:10]: This is something I don't like about blindly using IWYU. ```<cstddef>``` is what should be included. But for some reason IWYU includes the C header. One workaround for this that I've had some success with is replacing size_t with std::size_t for example.

File: app/actions.cpp [hunk 3]
```diff
@@ -1,49 +1,39 @@
...
-#include <sys/stat.h>                   // for stat()
+#include <sstream>
 #ifdef EXV_HAVE_UNISTD_H
-# include <unistd.h>                    // for stat()
+#include <unistd.h>  // for stat()
 #endif
 
 #if defined(_WIN32) || defined(__CYGWIN__)
-# include <sys/utime.h>
 #include <Windows.h>
```
Comment (neheb) [inline app/actions.cpp:33]: I thought windows.h was supposed to be lowercase.

File: include/exiv2/error.hpp (hunk 1 above)
Comment (neheb) [reply include/exiv2/error.hpp]: This seems to have been fixed in a later commit.

File: src/image_int.hpp (hunk 2 above)
Comment (piponazo) [reply src/image_int.hpp:10]: Right, I was not careful about this. I also prefer the usage of the <cXXX> headers. I will replace any occurrence of the C-Style headers by the C++ ones.

File: app/actions.cpp (hunk 3 above)
Comment (piponazo) [reply app/actions.cpp:33]: Yes, but then the NTFS filesystem is case-insensitive by default :sweat_smile: . Anyways, by googling it, I noticed that recently it is possible to enable the case sensitivity on Windows ... so I will correct this.

Comment (kmilos) [reply app/actions.cpp]: Btw, on MSYS2 (and I guess any other MinGW-w64 distribution), it really is `windows.h` :/

Comment (piponazo) [reply app/actions.cpp]: hum ... right ... somehow our CI was passing with MinGW & Cygwin ... I do not know if I want to dig there :sweat_smile:

**Comment by neheb** [comment]
LGTM

**Comment by piponazo** [comment]
> LGTM

Thanks! I need 1 approval to merge 😁 🙏

**Review by kmilos — APPROVED** [review]

**Pull request merged**